src/Kafka/CreateKafkaTopic/components/CreateTopicWizard.tsx

```diff
@@ -0,0 +1,183 @@
+import type React from "react";
+import { useState } from "react";
+import { useTranslation } from "react-i18next";
+import {
+  Divider,
+  PageSection,
+  PageSectionTypes,
+  PageSectionVariants,
+  Wizard,
+} from "@patternfly/react-core";
+import type { WizardStep } from "@patternfly/react-core";
+import {
+  StepMessageRetention,
+  StepPartitions,
+  StepReplicas,
+  StepTopicName,
+  WizardCustomFooter,
+} from "../components";
+import type { IWizardFooter } from "../components";
+import type { NewTopic } from "../types";
+import { PartitionLimitWarning } from "./PartitionLimitWarning";
+
+export type CreateTopicWizardProps = {
+  isSwitchChecked: boolean;
+  setIsCreateTopic?: (value: boolean) => void;
+  onCloseCreateTopic: () => void;
+  onSave: (topicData: NewTopic) => void;
+  initialFieldsValue: NewTopic;
+  checkTopicName: (value: string) => Promise<boolean>;
+  availablePartitionLimit: number;
+};
+
+export const CreateTopicWizard: React.FC<CreateTopicWizardProps> = ({
+  isSwitchChecked,
+  onCloseCreateTopic,
+  onSave,
+  initialFieldsValue,
+  checkTopicName,
+  availablePartitionLimit,
+}) => {
+  const { t } = useTranslation(["create-topic", "common"]);
+
+  const [topicNameValidated, setTopicNameValidated] = useState<
+    "error" | "default"
+  >("default");
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [invalidText, setInvalidText] = useState<string>("");
+  const [topicData, setTopicData] = useState<NewTopic>(initialFieldsValue);
+  const [warningModalOpen, setWarningModalOpen] = useState<boolean>(false);
+
+  const closeWizard = () => {
+    onCloseCreateTopic && onCloseCreateTopic();
+  };
+
+  const steps: WizardStep[] = [
+    {
+      name: t("topic_name"),
+      enableNext:
+        topicData?.name.trim() !== "" && topicNameValidated === "default",
```

Comment:
```suggestion
        topicData?.name.trim() && topicNameValidated === "default",
```
It will handle undefined and empty name.